src/data_snack/snack.py

```diff
@@ -38,14 +36,14 @@ def register_entity(self, cls: Type[Entity], key_fields: List[Text], serializer:
             raise WrongKeyValue("Entity keys cannot be empty")
 
         for key in key_fields:
-            if key not in cls.__dataclass_fields__.keys():
+            if key not in get_type_hints(entity_type).keys():
```

Comment:
Why do we need this change?

Reply:
We changed that in the other place and I forgot to change it here in the previous PR. We decided to use `get_type_hints` to avoid using dunder methods